delta.go

```diff
@@ -1208,41 +1226,32 @@ func (transaction *Transaction) PrepareCommit() (PreparedCommit, error) {
 
 // TryCommitLoop loads metadata from lock containing the latest locked version and tries to obtain the lock and commit for the version + 1 in a loop
 func (transaction *Transaction) TryCommitLoop(commit *PreparedCommit) error {
-	attemptNumber := 0
+	attempt := 0
 	for {
-		if attemptNumber > 0 {
+		if attempt > 0 {
 			time.Sleep(transaction.Options.RetryWaitDuration)
 		}
-		if attemptNumber > int(transaction.Options.MaxRetryCommitAttempts)+1 {
-			log.Debugf("delta-go: Transaction attempt failed. Attempts exhausted beyond max_retry_commit_attempts of %d so failing.", transaction.Options.MaxRetryCommitAttempts)
+		if attempt >= int(transaction.Options.MaxRetryCommitAttempts) {
+			log.Debugf("delta-go: Transaction attempt failed. Attempts exhausted beyond MaxRetryCommitAttempts of %d so failing.", transaction.Options.MaxRetryCommitAttempts)
 			return ErrExceededCommitRetryAttempts
```

Comment:
Might be useful to return an `errors.Join()` with the last error returned by `TryCommit()`

Reply:
Good point!